mobile/src/main/java/org/openhab/habdroid/background/ItemUpdateWorker.kt

```diff
@@ -91,34 +90,63 @@ class ItemUpdateWorker(context: Context, params: WorkerParameters) : Worker(cont
                     )
                     return@runBlocking Result.failure(buildOutputData(true, 500))
                 }
-                if (value == "TOGGLE") {
-                    value = determineOppositeState(item)
-                    mappedValue = value
+
+                val value = inputData.getValueWithInfo(INPUT_DATA_VALUE)!!
```

Comment:
Why move this inside the blocking block? If it's outside `valueToBeSent` could be logged in a failure.

Reply:
Because `valueToBeSent` needs `item` to be determined.

Reply:
Makes sense, but I like to log the value in case of an error. Especially for cases where the value is in an inappropriate format for the item, e.g. a badly formatted DateTime string.
I added the logging.